internal-fs/opt/kexec/check_kexec

```diff
@@ -16,18 +16,28 @@ else
     BBL_SN=$(bbl_3dpsn 2> /dev/null)
 fi
 
+# Set quickboot var
+QUICKBOOT_FLAG="false"
+if [ -f /mnt/sdcard/x1plus/printers/${BBL_SN}/settings.json ]; then
+    # Rely on settings.json if we can, default to false
+    QUICKBOOT_FLAG=$(/opt/kexec/jq ".\"boot.quick_boot\" // false" /mnt/sdcard/x1plus/printers/${BBL_SN}/settings.json)
+elif [ -f /mnt/sdcard/x1plus/printers/${BBL_SN}/quick-boot ]; then
+    # fallback to legacy flagfile check
+    QUICKBOOT_FLAG="true"
+fi
+
 # Is our quick boot flag set?
-if [ -f /mnt/sdcard/x1plus/printers/$BBL_SN/quick-boot ]; then
+if [ "${QUICKBOOT_FLAG}" == "true" ]; then
     # Before we do, are there any NEW x1p files?!
-    QUICK_BOOT=0 # Default to enable
+    NO_NEW_X1P=0 # Default to enable quick boot (0)
     for x1p in $(ls -1 /mnt/sdcard/*.x1p | sed -e 's|/mnt/sdcard/||g' | sed -e 's|.x1p||g'); do
         # Doesn't exist, bomb out
         if [ ! -d "/mnt/sdcard/x1plus/kernel/${x1p}" ]; then
-            QUICK_BOOT=1 # Disable
+            NO_NEW_X1P=1 # Disable
         fi
     done
     # If we are good to go, quick boot.
-    if [ "${QUICK_BOOT}" -eq "0" ]; then
+    if [ "${NO_NEW_X1P}" -eq "0" ]; then
```

Comment:
the inverted logic sense of this 1 vs. 0 makes me confused but I guess we did not really change it here otherwise.

Reply:
Was doing 0/1 cuz bash return codes, but yes it's kinda confusing, ill leave for now